frontend/src/components/EmailConfirmation.tsx

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from 'react';
+import axios from "axios";
+import { useParams } from "react-router-dom";
+
+function EmailConfirmation() {
+    const { id } = useParams<{ id: string }>();
+    const [confirmed, setConfirmed] = useState({confirmed: false})
+
+    useEffect(() => {
+        axios.put(`http://localhost:3000/users/${id}`, {confirmed: true}).then(res => {
+            setConfirmed({confirmed: res.data.confirmed});
```

Comment:
Likewise, once you update the initial state, you'll call:
`setConfirmed(res.data.confirmed)`